### PR TITLE
feat(datetime): add showDefaultTimeLabel prop

### LINKED ISSUE #23555: feat: Add option to remove "Time" Label from DateTime
# Feature Request

**Ionic version:**

[ ] **4.x**
[ ] **5.x**
[X] **6.x**

**Describe the Feature Request**

The DateTime Picker has a "Time" Label next to the Time selector. As i mostly want to use the Datetime Component inline like native ios, it would be great to be able to remove this text, as it isn't there on native ios either (compare reminder app).

**Describe Solutions**

- `timeLabel` Property which can be a string
- `showTimeLabel` boolean Property

I can create a PR for this if you say me which is the preferred way of handling this

### PR DESCRIPTION
**Pull request by EinfachHans**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [X] Build (`npm run build`) was run locally and any changes were pushed
- [X] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

The Label of the Time Selector of the datetime component was hardly set to "Time":

Issue Number: resolves #23555 

## What is the new behavior?

Changes done to match logic with title or buttons:
- Property `showDefaultTimeLabel` added
- Text is editable via the `time-label` slot

## Does this introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

**Comment by EinfachHans** [comment]
don't know why the test failed 🤔

**Review by liamdebeasi — CHANGES_REQUESTED** [review]
Thanks for the PR! Can you remove the `core/package-lock.json` change? Also can you add some tests for this? My initial thought is to add a `tests/time-label` directory that shows a datetime with `showDefaultTimeLabel="true"`, a datetime with `showDefaultTimeLabel="false"`, and a datetime with a custom time label slot.

I would use the `minmax` test as a starting point.

File: core/src/components/datetime/test/time-label/index.html [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        <div>
+            <h2>showDefaultTimeLabel="true"</h2>
+            <ion-datetime show-default-time-label="true"></ion-datetime>
+        </div>
+        <div>
+            <h2>showDefaultTimeLabel="false"</h2>
+            <ion-datetime show-default-time-label="false"></ion-datetime>
+        </div>
+      <div>
+        <h2>custom slot text</h2>
```
Comment (liamdebeasi) [inline core/src/components/datetime/test/time-label/index.html:46]:
```suggestion
        <h2>Custom Time Label</h2>
```

File: core/src/components/datetime/test/time-label/index.html [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+<html lang="en" dir="ltr">
+<head>
+    <meta charset="UTF-8">
+    <title>Datetime - Time label</title>
+    <meta name="viewport"
+          content="width=device-width, initial-scale=1.0, minimum-scale=1.0">
+    <link href="../../../../../css/ionic.bundle.css" rel="stylesheet">
+    <link href="../../../../../scripts/testing/styles.css" rel="stylesheet">
+    <script src="../../../../../scripts/testing/scripts.js"></script>
+    <script type="module" src="../../../../../dist/ionic/ionic.esm.js"></script>
```
Comment (liamdebeasi) [inline core/src/components/datetime/test/time-label/index.html:11]: Please use 2 space indentation here

**Review by liamdebeasi — CHANGES_REQUESTED** [review]
Can we add an `e2e.ts` file too that takes a screenshot so we can have a check for visual regressions? You can copy the file in `minmax`, just update the page url to point to `time-label`.

**Comment by EinfachHans** [comment]
@liamdebeasi Done. Should be exactly like the **minmax** test now from space indentation 🤔

File: core/src/components/datetime/datetime.tsx [hunk 3]
```diff
@@ -60,6 +60,7 @@ import {
  *
  * @slot title - The title of the datetime.
  * @slot buttons - The buttons in the datetime.
+ * @slot time-label - The label for the time selector in the datetime
```
Comment (liamdebeasi) [inline core/src/components/datetime/datetime.tsx:63]:
```suggestion
 * @slot time-label - The label for the time selector in the datetime.
```

**Review by liamdebeasi — CHANGES_REQUESTED** [review]
One small punctuation fix (you will need to do `npm run build` again), and we need to fix the indentation in the HTML test but otherwise this is looking good

**Comment by EinfachHans** [comment]
The indentation should be now the same as in the minmax test, isn't that the correct one?

**Comment by liamdebeasi** [comment]
The indentation is not the same.

Here is the indentation in the `minmax` file: 

image

Here is the indentation in the `time-label` file:

image

Do you have prettier enabled in your IDE? If so, I recommend turning that off as it may be automatically indenting files on save.

**Comment by EinfachHans** [comment]
Damn yes i had such a feature enabled, sorry. Thanks for finishing! 😊

**Review by liamdebeasi — APPROVED** [review]
I found a few other changes I wanted to make so I fixed the indentation issue in my commit as well. I think this is good to go - great job!

**Comment by liamdebeasi** [comment]
Merged. Thank you!!

**Comment by pierresh** [comment]
Hello,

I think it would worth adding an example in the documentation about how to change the time label. _cancelText_, _clearText_, and _doneText_ can be changed easily, but I could not find any example on internet about how to deal with time label.

**Pull request merged**